src/views/TabView/TabBarBottom.js

```diff
@@ -157,6 +164,30 @@ class TabBarBottom extends React.PureComponent<Props> {
       this.props.getTestIDProps && this.props.getTestIDProps(scene);
     return testIDProps;
   };
+  shouldUseHorizontalTabs() {
+    const { routes } = this.props.navigation.state;
+    const {
+      isLandscape,
+      initialLayout,
+      layout,
+      enableHorizontalTabs,
+    } = this.props;
+    const maxTabBarItemWidth = 125;
```

Comment:
Probably also read the `tabStyle` prop and use the `width` or `maxWidth` property if provided

Reply:
Checking for width works, but flow check can't handle `maxWidth` as a property of `tabStyle`. I get errors like `property 'maxWidth'. Property not found in Array|Boolean
Number.`

Reply:
You need to flatten the style property with `StyleSheet.flatten` before reading properties on it. Also, the properties can be percentage strings, in which case, you need to calculate the numeric value.